include/kafka.h

```diff
@@ -14,22 +14,27 @@
 #include <vector>
 #include <reading.h>
 #include <rdkafka.h>
+#include <config_category.h>
 
 /**
  * A wrapper class for a simple producer model for Kafka using the librdkafka library
  */
 class Kafka
 {
 	public:
-		Kafka(const std::string& brokers, const std::string& topic);
+		Kafka(ConfigCategory*& configData );
 		~Kafka();
 		uint32_t		send(const std::vector<Reading *> readings);
 		void			pollThread();
 		void			sendJSONObjects(bool arg) { m_objects = arg; };
 		inline void		success() { m_sent++; };
 		static void 		logCallback(const rd_kafka_t *rk, int level, const char *facility, const char *buf);
-
+		
 	private:
+		void			applyConfig_Basic(ConfigCategory*& configData);
+		void			applyConfig_SASL_PLAINTEXT(ConfigCategory*& configData, const std::string& kafkaSecurityProtocol);
+		void			applyConfig_SSL(ConfigCategory*& configData, const std::string& kafkaSecurityProtocol);
```

Comment:
signature of both are the same. Do you mean applyConfig and overload it with securityProtocol and saslMechanism?